ompi/mca/coll/base/coll_base_dynamic_file.c

```diff
@@ -0,0 +1,302 @@
+                        goto on_file_error;
+                    }
+                    msg_p->result_segsize = SS;
+
+                    if (!nms && MS) {
+                        opal_output_verbose(1,ompi_coll_base_framework.framework_output,"All algorithms must specify a rule for message size of zero upwards always first!\n");
+                        opal_output_verbose(1,ompi_coll_base_framework.framework_output,"Message size was %lu for collective ID %ld com rule %ld msg rule %d at around line %d\n", MS, CI, ncs, nms, fileline);
+                        ret = -7;
+                        goto on_file_error;
+                    }
+
+                    total_msg_count++;
+
+                } /* msg size */
+
+                total_com_count++;
+
+            } /* comm size */
+
+            total_nodes_count++;
+
+        } /* nodes number */
+        total_alg_count++;
+        opal_output_verbose(100,ompi_coll_base_framework.framework_output, "Done reading dynamic rule for collective ID %ld\n", CI);
+
+    } /* per collective */
+
+    ret = total_alg_count;
+    fclose (fptr);
+
+    opal_output_verbose(10,ompi_coll_base_framework.framework_output,"\nConfigure file Stats\n");
+    opal_output_verbose(10,ompi_coll_base_framework.framework_output,"Collectives with rules\t\t\t: %5d\n", total_alg_count);
+    opal_output_verbose(10,ompi_coll_base_framework.framework_output,"Nodes count with rules\t\t\t: %5d\n", total_nodes_count);
+    opal_output_verbose(10,ompi_coll_base_framework.framework_output,"Communicator sizes with rules\t\t: %5d\n", total_com_count);
+    opal_output_verbose(10,ompi_coll_base_framework.framework_output,"Message sizes with rules\t\t\t: %5d\n", total_msg_count);
+    opal_output_verbose(10,ompi_coll_base_framework.framework_output,"Lines in configuration file read\t\t: %5d\n", fileline);
+
+    /* return the rules to the caller */
+    *rules = alg_rules;
+
+    return ret;
+
+
+ on_file_error:
+
+    /* here we close out the file and delete any memory allocated nicely */
+    /* we return back a verbose message and a count of -1 algorithms read */
+    /* draconian but its better than having a bad collective decision table */
+
+    opal_output_verbose(1,ompi_coll_base_framework.framework_output,"read_rules_config_file: bad configure file [%s] which considered format is %d. Read afar as line %d\n"
+                                                , fname, format_version, fileline);
+    opal_output_verbose(1,ompi_coll_base_framework.framework_output,"Ignoring user supplied base collectives configuration decision file.\n");
+    opal_output_verbose(1,ompi_coll_base_framework.framework_output,"Switching back to [compiled in] fixed decision table.\n");
+    opal_output_verbose(1,ompi_coll_base_framework.framework_output,"Fix errors as listed above and try again.\n");
+
+    /* deallocate memory if allocated */
+    if (alg_rules) ompi_coll_base_free_all_rules (alg_rules, n_collectives);
```

Comment:
Minor style nit: please put all blocks in `{}`, even 1-line blocks.

We have very few style restrictions in Open MPI, but this is one of them.  Thanks!

(If you haven't already, you might want to check out https://github.com/open-mpi/ompi/wiki/CodingStyle)